code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -23,18 +23,17 @@ var/obj/item/weapon/disk/nuclear/nukedisk
 
 /obj/machinery/nuclearbomb/New()
 	..()
-	r_code = "[rand(10000, 99999.0)]"//Creates a random code upon object spawn.
+	r_code = "[rand(10000, 99999)]"//Creates a random code upon object spawn.
 
 /obj/machinery/nuclearbomb/process()
-	if (src.timing)
+	if(timing)
 		bomb_set = 1 //So long as there is one nuke timing, it means one nuke is armed.
-		src.timeleft--
-		if (src.timeleft <= 0)
+		timeleft -= SS_WAIT_MACHINERY/10 //Machinery does NOT tick every 10 ms. Divided by ten to convert into seconds
```

Comment:
I think this comment is misleading.
Should say something like, `// SS_WAIT_MACHINERY is given in deciseconds, we convert it in seconds here.`

Reply:
Perhaps, the original idea was to point out it shouldn't just be substracted by 1